internal/reconciliations/istio/configuration.go

```diff
@@ -60,131 +60,17 @@ func EvaluateIstioCRChanges(istioCR operatorv1alpha1.Istio, istioTag string) (tr
 		trigger = trigger | VersionUpdate
 	}
 
-	if nilChange(lastAppliedConfig.Components, istioCR.Spec.Components) {
+	if !cmp.Equal(lastAppliedConfig.Components, istioCR.Spec.Components) {
```

Comment:
I was not aware of the existence of this lib. I like this so much better, good job!